src/target/source/intrin_rule_cuda.cc

```diff
@@ -91,6 +106,19 @@ static void DispatchCUDAShuffle(const TVMArgs& args, TVMRetValue* rv) {
       call->dtype, "__shfl", cuda_args, CallNode::PureExtern);
 }
 
+static void DispatchCUDAShuffleSync(const TVMArgs& args, TVMRetValue* rv) {
+  PrimExpr e = args[0];
+  const CallNode* call = e.as<CallNode>();
+  CHECK(call != nullptr);
+  CHECK_EQ(call->args.size(), 4);  // value, warp_id/offset, width, warp_size
+
+  // mask is ignored.
+  const std::string& name = CUDAShuffleSync()(call->name);
+  auto mask = IntImm(DataType::UInt(32), 0xFFFFFFFF);
```

Comment:
Can we reserve `mask` as an argument? Since it is critical when there are branch divergence. `0xFFFFFFFF` is only a special case.